src/Symfony/Component/DependencyInjection/Tests/Loader/XmlFileLoaderTest.php

```diff
@@ -207,6 +208,20 @@ public function testParsesTags()
         }
     }
 
+    public function testParseTagsWithoutName()
+    {
+        try {
+            $container = new ContainerBuilder();
+            $loader = new XmlFileLoader($container, new FileLocator(self::$fixturesPath.'/xml'));
+            $loader->load('invalid_tag.xml');
+
+            $this->fail('The XmlFileLoader is expected to throw an exception when the name attribute is missing or a tag.');
+        } catch (InvalidArgumentException $e) {
+            $this->assertInstanceOf('\InvalidArgumentException', $e->getPrevious());
```

Comment:
I propose to throw the previous exception here, and use a phpunit annotation instead of all the assertions here